pkg/lib/bundle/validate.go

```diff
@@ -132,11 +170,159 @@ func (i imageValidator) ValidateBundle(directory string) error {
 
 	_, err = ValidateChannelDefault(annotations[ChannelsLabel], annotations[ChannelDefaultLabel])
 	if err != nil {
-		annotationErrors = append(annotationErrors, err)
+		validationErrors = append(validationErrors, err)
+	}
+
+	if len(validationErrors) > 0 {
+		return NewValidationError(validationErrors)
+	}
+
+	return nil
+}
+
+// ValidateBundleContent confirms that the CSV and CRD files inside the bundle
+// directory are valid and can be installed in a cluster. Other GVK types are
+// also validated to confirm if they are "kubectl-able" to a cluster meaning
+// if they can be applied to a cluster using `kubectl` provided users have all
+// necessary permissions and configurations.
+// Inputs:
+// manifestDir: the directory which all bundle manifests files are located
+// Outputs:
+// error: ValidattionError which contains a list of errors
+func (i imageValidator) ValidateBundleContent(manifestDir string) error {
+	var validationErrors []error
+
+	i.logger.Debug("Validating bundle contents")
+
+	mediaType, err := GetMediaType(manifestDir)
+	if err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	switch mediaType {
+	case HelmType, PlainType:
+		return nil
+	}
+
+	supportedTypes := map[string]string{
+		csvKind:                "",
+		crdKind:                "",
+		secretKind:             "",
+		clusterRoleKind:        "",
+		clusterRoleBindingKind: "",
+		serviceKind:            "",
+		serviceAccountKind:     "",
+		roleKind:               "",
+		roleBindingKind:        "",
```

Comment:
This will also need to include the prometheus types, see: https://github.com/operator-framework/operator-lifecycle-manager/pull/1227

Reply:
can we export these somehow rather than hardcoding them?

Reply:
even if that is in the API package and not in the registry? the more things like this that we start adding everywhere the harder it is going to be to change them.

Reply:
At the moment, there is nothing from OLM or registry to export. For OLM, we use a switch statement to handle all supported types. What OLM supports is a selected set of kube objects so there is no standard or logic for how we pick and choose.
I can create some struct with supported types and a function that validates if the type is supported. However, it is still some sort of "hardcoded".

Reply:
sure yeah, I undestand that. i mostly just mean now we have to keep track of this in two places and when the list updates remember to update them in both. what happens when the API project wants its own version of this validation for static analysis outside the scope of pulling down the image? Then the list is in 3 places.

not sure if we can solve that here right now, but i think this is a problem that we are facing sooner rather than later for a lot of things that are going to go across all of these projects. what is our import relationship between all of these repositories? or are we trying to explicitly avoid that completely by reimplementing the apis everywhere?

Reply:
I agree. There is a need to consolidate some agreements between OLM and registry. I would expect this change to begin with OLM. OLM needs to define a set of kube objects that it is supporting in a way that registry can import and then registry will enforce it.